lib/onelogin/ruby-saml/response.rb

```diff
@@ -51,6 +53,21 @@ def initialize(response, options = {})
 
         @response = decode_raw_saml(response)
         @document = XMLSecurity::SignedDocument.new(@response, @errors)
+
+        return unless assertion_encrypted?
```

Comment:
I created a method for that: decrypt_assertion_from_document    but the problem is that I need to clone the document in order to later decrypt it (if I no clone, then when decrypting I change the original object). And due Marshal bugs on  Ruby 1.8.7 I need to offer an alternative re-parsing the response.  And I want to pass to the method the cloned object. Does it make sense?